### ISSUE TITLE
Messaging: `publish` operation type is pub/sub/topic specific 

### ISSUE THREAD

**Issue by lmolkova** [open-telemetry/semantic-conventions#1285]
Extracted from https://github.com/open-telemetry/semantic-conventions/issues/1205

`messaging.operation.type = publish` is specific to pub/sub/topic.

`send` would sound more neutral and apply to pub/sub and MQ. E.g. GCP pub/sub, Azure EventHubs and Kafka use `send`.

We should consider changing operation type and also metric name to use `send`.

**Comment by mamwl1** [comment]
The JMS spec, https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.pdf, also uses send.
For example see: https://jakarta.ee/specifications/platform/8/apidocs/javax/jms/jmsproducer which has a set of send and receive methods which apply to both queues and topics.

**Comment by pyohannes** [comment]
> E.g. GCP pub/sub, Azure EventHubs and Kafka use send.

Ironically, [RabbitMQ prefers the term "publish"](https://www.rabbitmq.com/docs/publishers) 😄 .

I'm not generally opposed to using the term "send", however, I think we should be consistent.

Currently we give [this definition](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md) in semantic conventions, establishing the term for messaging semantic conventions:

> "Publishing" is the process of sending a message or batch of messages to the intermediary or consumer.

Do you think we should then generally revise our use of the term "publish" in [messaging semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md), and go with "send" instead?

**Comment by mamwl1** [comment]
Yes, I was going to propose adjusting the docs to use send generally instead of publish.
